### PR TITLE
[thread-cert] refactor case 9.2.7 using pktverify

### PR DESCRIPTION
**Pull request by FiveDimensions**

### CODE REVIEW

**Comment by simonlingoogle** [comment]
9.2.7 seems failed  in github action?

File: tests/scripts/thread-cert/Cert_9_2_07_DelayTimer.py [hunk 1]
```diff
@@ -129,12 +133,99 @@ def test(self):
...
+        # Step 4: Leader MUST send a unicast MLE Child ID Response to the Router
+        _lpkts.filter_wpan_dst64(ROUTER).filter_mle_cmd(MLE_CHILD_ID_RESPONSE).must_next(
+        ).must_verify(lambda p: {ACTIVE_OPERATION_DATASET_TLV, ACTIVE_TIMESTAMP_TLV} < set(p.mle.tlv.type) and {
+            NM_CHANNEL_TLV, NM_CHANNEL_MASK_TLV, NM_EXTENDED_PAN_ID_TLV, NM_NETWORK_MASTER_KEY_TLV,
+            NM_NETWORK_MESH_LOCAL_PREFIX_TLV, NM_NETWORK_NAME_TLV, NM_PAN_ID_TLV, NM_PSKC_TLV, NM_SECURITY_POLICY_TLV
+        } <= set(p.thread_meshcop.tlv.type) and p.mle.tlv.active_tstamp == LEADER_ACTIVE_TIMESTAMP)
+
+        # Step 6: Leader automatically sends a MGMT_ACTIVE_SET.rsp to the Router
+        _lpkts.filter_ipv6_dst(ROUTER_RLOC).filter_coap_ack(MGMT_ACTIVE_SET_URI).must_next().must_verify(
+            lambda p: p.coap.code == 68 and p.thread_meshcop.tlv.state == 1)
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_9_2_07_DelayTimer.py:170]: define constant for 68 and status 1?

File: tests/scripts/thread-cert/Cert_9_2_07_DelayTimer.py [hunk 2]
```diff
@@ -129,12 +133,99 @@ def test(self):
...
+        # Step 10: Leader MUST send a unicast MLE Data Response to the Router
+        _lpkts.filter_wpan_dst64(ROUTER).filter_mle_cmd(MLE_DATA_RESPONSE).must_next(
+        ).must_verify(lambda p: {ACTIVE_OPERATION_DATASET_TLV, ACTIVE_TIMESTAMP_TLV} < set(p.mle.tlv.type) and {
+            NM_CHANNEL_TLV, NM_CHANNEL_MASK_TLV, NM_EXTENDED_PAN_ID_TLV, NM_NETWORK_MASTER_KEY_TLV,
+            NM_NETWORK_MESH_LOCAL_PREFIX_TLV, NM_NETWORK_NAME_TLV, NM_PAN_ID_TLV, NM_PSKC_TLV, NM_SECURITY_POLICY_TLV
+        } <= set(p.thread_meshcop.tlv.type) and p.mle.tlv.active_tstamp == ROUTER_ACTIVE_TIMESTAMP)
+
+        # Step 12: Leader sends a MGMT_PENDING_SET.rsp to the Router with Status = Accept
+        _lpkts_coap.filter_ipv6_dst(ROUTER_RLOC).filter_coap_ack(MGMT_PENDING_SET_URI).must_next().must_verify(
+            lambda p: p.coap.code == 68 and p.thread_meshcop.tlv.state == 1)
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_9_2_07_DelayTimer.py:188]: ditto

File: tests/scripts/thread-cert/Cert_9_2_07_DelayTimer.py [hunk 3]
```diff
@@ -129,12 +133,99 @@ def test(self):
...
+
+        # Step 13: Leader sends a multicast MLE Data Response
+        _lpkts.filter_LLANMA().filter_mle_cmd(MLE_DATA_RESPONSE).must_next().must_verify(
+            lambda p: {
+                SOURCE_ADDRESS_TLV, LEADER_DATA_TLV, ACTIVE_TIMESTAMP_TLV, PENDING_TIMESTAMP_TLV, NETWORK_DATA_TLV
+            } <= set(p.mle.tlv.type) and p.thread_nwd.tlv.stable == [0] and p.mle.tlv.active_tstamp ==
+            ROUTER_ACTIVE_TIMESTAMP and p.mle.tlv.pending_tstamp == ROUTER_PENDING_TIMESTAMP)
+
+        # Step 14: The DUT MUST send MGMT_DATASET_CHANGED.ntf to the Router
+        _lpkts_coap.filter(lambda p: p.wpan.dst16 == ROUTER_RLOC16).filter_coap_request(
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_9_2_07_DelayTimer.py:198]:
```suggestion
        _lpkts_coap.filter_wpan_dst16(ROUTER_RLOC16).filter_coap_request(
```
add `filter_wpan_dst16` ?

File: tests/scripts/thread-cert/Cert_9_2_07_DelayTimer.py [hunk 4]
```diff
@@ -129,12 +133,99 @@ def test(self):
...
+            MGMT_DATASET_CHANGED_URI).must_next()
+
+        # Step 16: Leader MUST send a unicast MLE Data Response to the Router
+        _lpkts.filter_wpan_dst64(ROUTER).filter_mle_cmd(MLE_DATA_RESPONSE).must_next().must_verify(
+            lambda p: {ACTIVE_TIMESTAMP_TLV, PENDING_TIMESTAMP_TLV} < set(p.mle.tlv.type) and p.mle.tlv.active_tstamp
+            == ROUTER_ACTIVE_TIMESTAMP and p.mle.tlv.pending_tstamp == ROUTER_PENDING_TIMESTAMP)
+
+        # Step 18: The DUT MUST send MGMT_PENDING_SET.rsp to the Commissioner
+        _lpkts_coap.filter_ipv6_dst(COMMISSIONER_RLOC).filter_coap_ack(MGMT_PENDING_SET_URI).must_next().must_verify(
+            lambda p: p.coap.code == 68 and p.thread_meshcop.tlv.state == 1)
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_9_2_07_DelayTimer.py:208]: ditto

File: tests/scripts/thread-cert/Cert_9_2_07_DelayTimer.py [hunk 5]
```diff
@@ -129,12 +133,99 @@ def test(self):
...
+                SOURCE_ADDRESS_TLV, LEADER_DATA_TLV, NETWORK_DATA_TLV, ACTIVE_TIMESTAMP_TLV, PENDING_TIMESTAMP_TLV
+            } <= set(p.mle.tlv.type) and p.thread_nwd.tlv.stable == [0] and p.mle.tlv.active_tstamp ==
+                                                       ROUTER_ACTIVE_TIMESTAMP and p.mle.tlv.pending_tstamp == 40)
+
+        # Step 20: Leader MUST send a unicast MLE Data Response to the Router
+        _lpkts.filter_wpan_dst64(ROUTER).filter_mle_cmd(MLE_DATA_RESPONSE).must_next(
+        ).must_verify(lambda p: {ACTIVE_TIMESTAMP_TLV, PENDING_TIMESTAMP_TLV, PENDING_OPERATION_DATASET_TLV} < set(
+            p.mle.tlv.type) and {NM_CHANNEL_TLV, NM_COMMISSIONER_SESSION_ID_TLV, NM_PAN_ID_TLV, NM_DELAY_TIMER_TLV
+                                } <= set(p.thread_meshcop.tlv.type) and p.mle.tlv.active_tstamp ==
+                      ROUTER_ACTIVE_TIMESTAMP and p.mle.tlv.pending_tstamp == 40 and p.thread_meshcop.tlv.pan_id ==
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_9_2_07_DelayTimer.py:222]: define constant for 40?

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 
 with nits

File: tests/scripts/thread-cert/Cert_9_2_07_DelayTimer.py (hunk 1 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_9_2_07_DelayTimer.py:170]: 👌

File: tests/scripts/thread-cert/Cert_9_2_07_DelayTimer.py (hunk 3 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_9_2_07_DelayTimer.py]: added

**Review by jwhui — APPROVED** [review]

**Pull request merged**